src/middleware.rs

```diff
@@ -68,6 +67,11 @@ pub fn build_middleware(app: Arc<App>, endpoints: RouteBuilder) -> MiddlewareBui
     m.add(AppMiddleware::new(app));
     m.add(KnownErrorToJson);
 
+    if env != Env::Test {
+        // This is added *after* AppMiddleware to make sure the app is available.
+        m.add(log_request::LogRequests::default());
```

Comment:
Moving this line will change what we log. In particular, I'm worried that it will log requests before normalizing well-known errors into JSON. It would also log paths after normalization, instead of the original client request.